inspirehep/modules/migrator/tasks.py

```diff
@@ -356,3 +359,72 @@ def migrate_and_insert_record(raw_record):
         prod_record.valid = True
         db.session.merge(prod_record)
         return record
+
+
+@shared_task(ignore_result=True)
+def sync_legacy_claims():
+    """Task to import claims from legacy."""
+    redis_url = current_app.config.get('CACHE_REDIS_URL')
+    r = StrictRedis.from_url(redis_url)
+    lock = Lock(r, 'import_legacy_claims', expire=120, auto_renewal=True)
+    if lock.acquire(blocking=False):
+        try:
+            while r.llen('legacy_claims'):
+                claim = loads(r.lrange('legacy_claims', 0, 1)[0])
+                apply_claim(*claim)
+                r.lpop('legacy_claims')
+        finally:
+            lock.release()
+    else:
+        logger.info('Import_legacy_claims already executed. Skipping.')
```

Comment:
I think this should be at least an error, so it shows up on sentry, @kaplun it should not happen right? and if it does, we should actually take a look.

I think that's better than ignoring it completely, as we would not notice that it got stuck.

Reply:
Nope it can happily happen. When the currently executing one is too slow to operate. E.g. say that it take 10 minutes to ingest all the latest claims (very improbable), and that the task is scheduled to run every minute in celery beat: then after 1 minute a second identical task will be executed and we don't want that. But it's not an error. It's just the system being unnecessarily slow...